internal/transport/http2_server.go

```diff
@@ -1320,6 +1329,9 @@ func (t *http2Server) outgoingGoAwayHandler(g *goAway) (bool, error) {
 		t.state = draining
 		sid := t.maxStreamID
 		if len(t.activeStreams) == 0 {
+			if logger.V(logLevel) {
+				logger.Infof("transport: second goaway written and no active streams left to process, closing the connection.")
```

Comment:
This just exits loopy writer. We're logging when loopy exits already, so this shouldn't be necessary (maybe refine error message is all).

Reply:
Ah fair enough, this is the wrong language. This triggers loopy to exit, which logs the error/closes conn downstream. However, there's a few ways the bool g.closeConn gets set to true, which at the loopyWriter level all see the same thing (closeConn set to true). How would you be able to refine the error message to give info as to why loopy is exiting if at that layer it only sees that closeConn was set to true?

Reply:
One option: change `g.closeConn` to an `error` type and write something nicer in there wherever it's set to true elsewhere.  Then:

```go
retErr := g.closeConn
if len(t.activeStreams) == 0 {
  retErr = fmt.Errorf("transport: loopy exiting due to second GOAWAY with no active streams or whatever")
}
// ....
if retErr {
  flush(); return false, retErr
}
```